straxen/plugins/multi_scatter.py

```diff
@@ -0,0 +1,325 @@
+import numba
+from straxen.plugins.position_reconstruction import DEFAULT_POSREC_ALGO
+from straxen.common import pax_file, get_resource, first_sr1_run, rotate_perp_wires
+from straxen.get_corrections import get_cmt_resource, is_cmt_option
+from straxen.itp_map import InterpolatingMap
+import strax
+import straxen
+import numpy as np
+export, __all__ = strax.exporter()
+
+@export
+class EventPeaks(strax.Plugin):
+    """
+    Add event number for peaks and drift times of all s2 depending on the largest s1.
+    Link - https://xe1t-wiki.lngs.infn.it/doku.php?id=weiss:analysis:ms_plugin
+    """
+    __version__ = '0.0.57'
+    depends_on = ('event_info', 'peak_basics','peak_positions')
+    provides = 'peaksevent'
+    data_kind = 'peaks'
+
+
+    
+    def infer_dtype(self):
+        dtype = []
+        dtype += [
+            ((f'event_number'), np.float32),
+            ((f'drift_time'), np.float32),
+            ((f'correction'), np.float32),
+             ]
+        dtype += strax.time_fields
+        return dtype
+
+    save_when = strax.SaveWhen.TARGET
+ 
+        
+    def compute(self, events, peaks):
+        EB = straxen.EventBasics()      
+        split_peaks = strax.split_by_containment(peaks, events)
+        split_peaks_ind = strax.fully_contained_in(peaks, events)
+        result = np.zeros(len(peaks), self.infer_dtype())
+        result.fill(np.nan)
+        #result['s2_sum'] = np.zeros(len(events))
+   
+         #1. Assign peaks features to main S1 and main S2 in the event
+        for event_i, (event, sp) in enumerate(zip(events, split_peaks)):
+            result[f'drift_time'][(split_peaks_ind==event_i)] = sp[f'center_time']-event[f's1_center_time']
+        result[f'event_number']= split_peaks_ind
+        result[f'drift_time'][peaks["type"]!=2]= np.nan
```

Comment:
can't we pre-select s2 peaks earlier?

Reply:
The problem is that I am using result which is the size of peaks . I f I pre select have trouble putting the drift time in the correct location. If I decide on removing all s1 but the main one this will work. However this will not be peak data anymore.